Remove 'key' in URL so make the request valid on Google's API

pascalfleury opened the pull request:
org-gtasks started to fail for me, with an error 400. Debugging the RPC led me to remove this key attribute. Now I can sync the tasks again.

Commit:
Improved the README to give better bootstrap information.

Commit:
Cleanup the markup.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Fix a typo.

Commit:
Avoid sending the secret around as a key, this is now an error on the Google API.

JulienMasson commented:
Hello, very sorry for the late reply.
I've been very busy on several projects in the same time.
To be honest with you I did not use org-gtasks the last few months...

But this week I started to re-use the module and indeed the "key" param is not needed anymore.
Thanks for catching that, I applied your patch into the master:
`669a3b3 Avoid sending the secret around as a key, this is now an error on the Google API.`

Also I took some time to review again the source code and different things annoyed me.
Especially all the requests were blocking which is pretty bad in term of user experience.
So I refactor all the code to make all requests async and fixed some bugs.

It's already in the master:
`28a7b2e [REWORK] make all requests async and fix several issues`

I tried to test all the cases that I had in mind.

Please let me know if you have issues with last changes :)

Thanks

pascalfleury commented:
hey thanks a lot!

I will sync to head then. I was using my modified version anyway, so no
problem.
these are busy times anyway...

take care,
--paf

Le mer. 25 nov. 2020 à 18:45, Julien Masson <notifications@github.com> a
écrit :

> Hello, very sorry for the late reply.
> I've been very busy on several projects in the same time.
> To be honest with you I did not use org-gtasks the last few months...
>
> But this week I started to re-use the module and indeed the "key" param is
> not needed anymore.
> Thanks for catching that, I applied your patch into the master:
> 669a3b3 Avoid sending the secret around as a key, this is now an error on
> the Google API.
>
> Also I took some time to review again the source code and different things
> annoyed me.
> Especially all the requests were blocking which is pretty bad in term of
> user experience.
> So I refactor all the code to make all requests async and fixed some bugs.
>
> It's already in the master:
> 28a7b2e [REWORK] make all requests async and fix several issues
>
> I tried to test all the cases that I had in mind.
>
> Please let me know if we have issues with last changes :)
>
> Thanks
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JulienMasson/org-gtasks/pull/4#issuecomment-733854853>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBP5KABQMSACHWWR7LM4QDSRU7BRANCNFSM4QQ6EFBA>
> .
>

pascalfleury committed:
fixed merge conflicts

pascalfleury committed:
Merge branch 'JulienMasson-master'

pascalfleury committed:
Add list name as optional to avoid having a user prompt.

This is very useful when you want to bind this in a call from a timer
to update the tests regularly (e.g. when refreshing the calendar view).

pascalfleury committed:
Typo and additional info about the refresh token in docs.

pascalfleury commented:
This is now working from head, I verified it.
